docs/gdpr.md

```diff
@@ -0,0 +1,20 @@
+# Tracking Consent
+
+### Tracking consent values
+
+In order to be compliant with the GDPR regulation the SDK needs requires the tracking consent value at initialization.
+The tracking consent can have one of the values:
+
+1. `TrackingConsent.PENDING` - the SDK will start collecting and batching the data but will not send it to the data 
+   collection endpoint. It will wait for the new tracking consent value to decide what to do with the batched data.
+2. `TrackingConsent.GRANTED` - the SDK will start collecting the data and will be sending it to the data collection endpoint.
+3. `TrackingConsent.NOT_GRANTED` - the SDK will not collect any data. You will not be able to manually send any logs, traces or 
```

Comment:
`not_granted` and `pending` states overlap logically
what do you think of `denied` instead of `not_granted`?

Reply:
What do you mean ? They are not the same and in any case it will create confusion if you want to name it `denied` as in the code is not like that.

Reply:
i mean `(granted) or (not granted)` must always be true in logic
[this is an example from iOS SDK](https://developer.apple.com/documentation/usernotifications/unauthorizationstatus): the status can be either `notDetermined` or one of `determined` states (ie: `denied`, `authorized`, `provisional`, `ephemeral`)

yes, if you decide to rename you should rename all in the project